### DISCUSSION TITLE
Connecting to my Fritzbox 7590, LILIYGO T-Beam V1.2 ESP32 Lora modul/ SX1278

### DISCUSSION THREAD

**Discussion by georgjorge** [dl9rdz/rdz_ttgo_sonde#434] [TTGO firmware]
I loaded two different firmware files (devel20240316-B17-full.bin and devel20240225-B17-full.bin) with the same result. I am able to connect directly via my smartphone S23, only to change the SSDI and network Passwort in Liliy´s firmware. After reboot and connecting the router, Lily shows an IP 192.168.178.78 on the display, but the router says not connected, timeout. The IP is removed from the display. How could the router be forced to connect?
Georgjorge

**Comment by georgjorge** [comment]
In addition to my post yesterday, my hardware is "LILIY T-Beam V1.2 ESP32 Lora Modul Ipex GPS-NEO-M8N SX1278 433MHz. Today I played around and sometimes I succeded to change the parameters, but often it happens that Lily lost it´s WLAN connection. Could it be that my hardware is not compatible with the firmware?
Georgjorge
dl2kp

**Reply by hr-ru** [reply]
I have observed one issue with the T-Beam 1.2 that I have not seen with any other boards. I am having WiFi issues as well (board works fine as AP but does not connect as station to an existing AP). This is happening only with this board. It is not happening if the bootloader is configured to use 40 MHz SPI flash mode instead of 80 MHz (which is the default in the binary images).

If you compile with Arduino IDE, this means you should select "Tools"->"Flash Frequency"->40 Mhz (at least that fixes my problem. I do not yet understand the cause of this problem, this is very strange)

**Comment by georgjorge** [comment]
Im my case, I tried several binary images with the ESP32 FLASH Download Tool V3.9.6 with 40MHz SPI Speed and SPI Mode DIO. I will try it again to compile with VS Code and platformio.

**Comment by georgjorge** [comment]
HI,
I am on the verge of giving up. My hardware is signed on the board as "TBeam-AXP2101_V1.2 20230508". The firmware loaded with 40MHz SPI speed is master-v0.9.4. I am unable to connect my Samsung S3 to the presented IP 192.168.4.1. The error message is "unreliable connection". Connecting via my router the message is "time out". The display attached to the board schows scanning through the various frequencies.
Further more, I set up Arduino IDE and compiled the source offered on github. The result was even worse. I got the spectrum without the IP, only. Maybe, I didn´t setup the IDE correctly. I have to check again.
My question is, should I buy another board for test, or is it likely to run into a similar situation?
Georgjorge
dl2kp

**Comment by georgjorge** [comment]
I tried the IDE-example WiFiClientConnect. With this testbed the testboard says it is connected and the Fritz!Box prints "Connection not successful, time out". Therefore I am not able to connect my Smartphone to the board. If anybody is familiar with the router software, I can send my board for testing. This could be of interest to other builders, too.
Vy 73, Georg
DL2KP

**Comment by hr-ru** [comment]
> and the Fritz!Box prints "Connection not successful, time out". 

Where do you see this message? Can you make a screenshot? Or explain where this is printed?

**Comment by georgjorge** [comment]
This is a message generated by Fritz!Box 7590:
Fritz!Box >WLAN>Funknetz:
- WPS starten > Die Push-Button-Methode ist aktiv > Message:"Die automatisierte Anmeldung (WPS) ist wegen Zeitüberschreitung fehlgeschlagen.

**Comment by hr-ru** [comment]
WPS ist nicht unterstützt. Das wäre nur eine Komfort-Anmeldung ohne dass man das WLAN-Passwort im TTGO eintragen müsste. Das ist nicht notwendig. Wenn im TTGO das WLAN-Passwort konfiguriert ist, dann verbindet sich das TTGO mit der Fritzbox. Da scheint mir alles richtig zu sein.

**Comment by georgjorge** [comment]
Dazu muß ich aber das Passwort setzen können. Ich habe folende Situation:
Firmware 0.941-B17-full mit 40MHz flash config auf board geladen. Reset. Auf dem Display erscheint die IP 192.168.4.1. Wenn jetzt mit meinem Samsung S23 und Firefox v. 125.0.2 (64Bit) eine Verbindung versucht wird, erscheint die Fehlmeldung auf dem Samsung "Verbindung fehlgeschlagen" und weiter noch "Verbindung ist nicht sicher". Außerdem wird angezeigt, daß der "Verbesserter Schutz vor Aktivitätenverfolgung" aktiviert ist. Er kann zwar ausgeschaltet werden, wird aber sofort automatisch eingeschaltet. 

Wird die IP im Brauser meines Win11 PC eingegeben, dann erhalte ich die Fehlermeldung : Seintn-Ladefehler und Der Server unter 192.168.4.1 braucht zu lange, um eine Antwort zu senden.

**Reply by hr-ru** [reply]
Ok, dann kommen wir der Sache schon näher.

Evtl auch siehe hier: https://www.wettersonde.net/rdzTTGOSonde-Installation.php#rdz_wlan

Solange kein WLAN/Passwort konfiguriert ist, läuft das TTGO als Access Point. D.h. man kann Handy oder Computer direkt damit verbinden, in dem man dort (Handy/Computer) sich mit dem WLAN mit Namen RDZsonde (Password RDZsonde) verbindet.  Dann kann man mit dem Browser auf http://192.168.4.1 zugreifen.

Als nächsten kann man dann im Tab "WiFi" die Zugangsdaten der eigenen Fritzbox eintragen (bei Nr. 1, man kann auch mehrere WLAN-Netze eintragen). Dann noch den WiFi-Modus ändern (unter Config) und TTGO neustarten, dann sollte sich das TTGO mit der Fritzbox verbinden, und im Display sollte eine andere IP-Adresse stehen.

**Comment by georgjorge** [comment]
Jetzt bin ich einen Schritt weiter.  Hier die Abfolge:

- Load firmware with 40MHz flash speed
- RESET board
- Go to setting on your smartphone
- WLAN
- Look for RDZsonde and klick on it
- Add to network
- Type name of network and password
- If you go back to WLAN you can see the actual network : RDZsonde without internet
- Now go to your browser and type the IP shown on the board´s display
- The browser shows a list of QRGs
- Klick on hamburger menu
- The available menus are schown
- Choose WiFi
- Add the SSID and Networkkey of your router
- Save changes
- Reset
- ????????????????????
Hier kommt wieder die alte IP 192.168.4.1. Irgend etwas mache ich noch falsch, aber was?

**Reply by hr-ru** [reply]
Hier gibt es evtl zwei Möglichkeiten
(a) Irgendwas ist falsch, z.B. Tippfehler bei den Daten
(b) Ich habe hier bei meinem T-Beam V1.2 das Problem, dass genau das nicht geht. Das tritt nur bei diesem einen Board auf, bei keinem anderen.  Die Ursache habe ich noch nicht verstanden, aber was bei mir hilft ist der 40 MHz-Modus für das SPI-Flash. Alllerdings ist das nicht das, was man beim Flashen im Flashtool einstellen kann, sondern das ist in der Firmware (im Bootloader) enthalten. Das kann man also nicht so einfach ändern (ausser durch Image neu erstellen).

**Comment by georgjorge** [comment]
Bingo, sobald man es richtig macht funktioniert es auch! Im SSID war ein Blank zuviel. Auch mit 81 Jahren schafft man es, wenn mann nicht aufgibt. Vielen Dank Hans, das war eine harte Nuß für mich. Jetzt muß ich es noch schaffen mein 3.5" Display zum Spielen zu bekommen. Das wird aber ein neuer Beitrag.
Ein schönes Wochenende wünscht
Georg alias Georgjorge,
DL2KP
